### PR TITLE
Adding github repository metadata 

### LINKED ISSUE #166: Github repository meta documents missing
The following documents are missing and should be added to validate the open source project and structure a community.

- [ ] Pull Request Template
- [ ] Issue Template
- [ ] Contributing Guidelines
- [ ] Code Of conduct

Working on it, claiming.

### PR DESCRIPTION
**Pull request by YashKumarVerma**
The following documents are missing and should be added to validate the open source project and structure a community.

- [x] Pull Request Template
- [x] Issue Template
- [x] Contributing Guidelines
- [x] Code Of conduct

Fixes #166

### CODE REVIEW

**Comment by YashKumarVerma** [comment]
> Warnings are found on analyzing the commit [846958d](https://github.com/coding-blocks/shortlr2-backend/commit/846958d1346507c72cc980768746ab1c2706ea1c).
> 
> **1** warning:
> 
> * warning [TSLint](https://sider.review/gh/repos/155214828/pulls/167/logs/f72c65b3-c122-4637-8894-0992cda99b92)
> 
> We recommend to address them as possible, for example, update outdated dependencies, fix the tool's configuration, [configure `sider.yml`](https://help.sider.review/getting-started/custom-configuration), [turn off unused tools](https://help.sider.review/getting-started/repository-settings), and so on.
> 
> If you are struggling with these errors or warnings, feel free to ask us via chat.

error not triggered by PR, cause by existing file

**Comment by championswimmer** [comment]
Hi @YashKumarVerma 

You have been making this generic PR of adding github templates across all your repos. 

So let me take this up at here in one place. 

1. not all projects need this. not all projects even need all four of these. CoC for org is enough, copying everywhere isn't needed
2. are you doing this for getting just some more bounties ? please do not do that. you wrote an email to me accusing others of doing the same, and here I see you doing this
3. writing points like `Lint and unit tests pass locally with my changes` in a project that has neither linter nor tests setup in the checklist.... what for ? 

Can we keep our focus on code that 

1. fixes bugs
2. adds features
3. improves coding standards
4. improves architecture of project
5. adds tests

anything else, administrative work, please leave it to us mentors to do. does that make sense ?

**Comment by YashKumarVerma** [comment]
Thanks for pointing out and mentioning on the discord server.
It would be helpful for me to correct/update the place where I did the following. If you could specifically point where, i can fix it.
> 3. writing points like `Lint and unit tests pass locally with my changes` in a project that has neither linter nor tests setup in the checklist.... what for ?

**Comment by championswimmer** [comment]
Do you not even read your own PR before making it? This very PR has a checklist that contains that point. 

What kind of irresponsible behaviour is this to make PRs whose contents you have no idea of.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- [ ] Bugfix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+## Checklist
+
+_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._
+
+- [ ] I have read the [CONTRIBUTING](/CONTRIBUTING.md) doc
+- [ ] Lint and unit tests pass locally with my changes
```
Comment (championswimmer) [inline .github/PULL_REQUEST_TEMPLATE.md:23]: Here @YashKumarVerma

**Comment by YashKumarVerma** [comment]
> Do you not even read your own PR before making it? This very PR has a checklist that contains that point.
> 
> What kind of irresponsible behaviour is this to make PRs whose contents you have no idea of.

Thank you for the **polite** reply. Really appreciate it.

> writing points like Lint and unit tests pass locally with my changes in a project that has neither linter nor tests setup in the checklist.... what for ?

image

I'd like to draw attention towards https://github.com/coding-blocks/shortlr2-backend/blob/master/package.json#L32 which has something along the same lines @championswimmer

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (YashKumarVerma) [reply .github/PULL_REQUEST_TEMPLATE.md:23]: more context provided as comment

**Comment by championswimmer** [comment]
Listen @YashKumarVerma if you want to get into a sarcasm match, feel free to go ahead. I will love to participate. I love Twitter/Github battles. It improves my vocabulary and conversation skills. 

Coming to real business. 

- Regarding this PR.
  - Ok my bad, linter is setup, but having a checklist point in the PR makes no sense when Sider will tell me if there are lint errors anyway. If someone marks that checkbox as done, even if lint fails, then ? If someone forgets marking it done, even if lint is passing then ? 
Also everytime you run the project locally, `lint:fix` already runs, so it is kinda hard to make a PR without lint passing isn't it ? 
  - tests are *NOT* really set up here. one test for one function in a 50 file project, with no coverage over the middlewares and controllers, and no e2e test setup at all isn't called "tests are setup" so that checkbox makes no sense
  - `Any dependent changes have been merged and published in downstream modules` Are there other submodules does this repo have that we need to merge changes into ? 

- In general, as I have already said, please do not think that going to every repo and doing administrative tasks is the easiest way to gain points. It is sad that you especially, among other participants (who has so much prior OSS experience) is making this about collecting points rather than focussing on doing some real work. for tasks like adding codecov integration, those accesses anyway are with mentors and they only can do it. also mentors are best to decide at what stage (scale, size) of project it starts making sense to have travis build each pr or codecov to auto comment on the PRs.

**Comment by YashKumarVerma** [comment]
### Follow a Code of Conduct
 
> Listen @YashKumarVerma if you want to get into a sarcasm match, feel free to go ahead. I will love to participate. I love Twitter/Github battles. It improves my vocabulary and conversation skills.

Sir, all I said was a simple think you for your choice of words, and you're accusing me of sarcasm. Yes, that was sarcasm, because I won't resort to harsh words, as you being my senior and language like this not something that people expect from open source maintainers. 

> Do you not even read your own PR before making it? This very PR has a checklist that contains that point.
> What kind of irresponsible behavior is this to make PRs whose contents you have no idea of.

especially when it was you who made a mistake there. I very respectfully made a valid point. What followed was an act of frustration. 

---

> Coming to real business.
> 
> * Regarding this PR.
>   
>   * Ok my bad, linter is setup, but having a checklist point in the PR makes no sense when Sider will tell me if there are lint errors anyway. If someone marks that checkbox as done, even if lint fails, then? If someone forgets marking it done, even if lint is passing then?

Exactly my point. 
I thought to add it might be a good thing, but it appears that it was not a good idea at all. Third-party integrations would surely point them out in the sidebar. I agree with it.

Regarding tests, yes currently the project does not have extensive tests written, I'm sure it would be done in the near future. Putting those checkboxes there was indeed a wrong attempt, same is the case with the modules line. 

This was expected in the first place instead of `Do you....have no idea of`. If you think this is not to be done, simply saying that it's admin decision and close it. But since it wasn't done that way, I need to reply to the statements made.

> * In general, as I have already said, please do not think that going to every repo and doing administrative tasks is the easiest way to gain points. It is sad that you especially, among other participants (who has so much prior OSS experience) is making this about collecting points rather than focussing on doing some real work. for tasks like adding codecov integration, those accesses anyway are with mentors and they only can do it. also mentors are best to decide at what stage (scale, size) of project it starts making sense to have travis build each pr or codecov to auto comment on the PRs.

+1 to it, but here's my opinion on it.

Sir I am not here to argue with you regarding whether this must be done by a non-admin or not. Neither am I requesting you to reopen this, nor to consider the changes or claiming for points. All these issues / prs were opened about 9-10 days back within a span of few minutes, about the changes that I thought can be implemented. 

There was no message or instruction that similar things are not to be done, (if there was, I'm sorry I must have mistakenly skipped it) so I opened these. My belief regarding this was, even if admins need to grant permissions and install 3rd party applications, someone needs to write the scripts for
it. I was maybe wrong at it. 

---

Now as far as BOSS is concerned, Have a look at this. https://github.com/coding-blocks/boss/issues/created_by/YashKumarVerma. These were again opened with the same date as this pr. 
image
Bounty was linked to only one issue, meanwhile, I sent PRs for all of them. 
People who are working on only issues marked for competition are not doing it for bounty, but if someone is sending PRs for issues not tagged for boss, they're doing it for bounty only. It's a bit confusing.

---
### Rejecting valid contribution in the sight of BOSS
On a side note, please do not consider that everything that's done is done for the competition as was done here: https://github.com/coding-blocks/oneauth/pull/994
Was this PR valid?
- Yes, .gitkeep files should be removed if the directory is not empty. Done in [Laravel](https://github.com/laravel/laravel/pull/4008)
Your reply was
> let's not make PRs just for the sake of BOSS bounties. work on productive things please.
So apparently we must overlook the minor issues.

You have all the rights to say so when I do things that don't actually improve the project, but you also know that those files are no longer required. Still? Did I ask for bounty points? No. But there have been similar cases, https://github.com/coding-blocks/boss/pull/303#issuecomment-630332641
- Can you mention a similar case when I asked for points?
- Can you mention any case when I made a claim in the portal?

---

Moving on, a few suggestions if you wish 
- be respectful.
- value every contribution, not everyone comes to OSS to write features and fix bugs.
- make users clearly mention in their PRs if it's for BOSS or not, else you'll keep rejecting valid changes.
- when complaints are raised, instead of trying to target the raiser, try to fix the issues raised.

**Comment by championswimmer** [comment]
> I'm sorry but I'm facing issues in searching for it. 

> Thank you for the **polite** reply. Really appreciate it. 
>> (emphasis on polite)

I think I need a Code of Conduct against snarky comments like this 😅

But since we do not have a code of conduct yet for this project, let me happily go ahead and drop a meme to close this discussion 
image

**Pull request closed without merging**